net: Rm discontinued passt

nirdothan opened the pull request:
<!--  Thanks for sending a pull request!  Here are some tips for you:
1. Consider creating this PR as a draft: https://github.com/kubevirt/kubevirt/blob/main/CONTRIBUTING.md#consider-opening-your-pull-request-as-draft
2. Follow the instructions for writing a release note from k8s: https://git.k8s.io/community/contributors/guide/release-notes.md
-->

### What this PR does
`passt` used to be a core network binding and was discontinued [1] in KubeVirt release 1.3,
as Network Binding Plugins functionality was introduced. 
Block attempts to create VMs with `passt`  binding, as was done with `slirp` binding.
In addition remove remainders passt code such as the (core) binding is entirely removed across the stack.
As such `passt` can no longer be used as a network binding. 
`passt` is available as network binding **plugin**.

Remove unused validations.

[1] https://github.com/kubevirt/kubevirt/pull/11915

#### Before this PR:

#### After this PR:

### References
<!-- optional,
  Use `Fixes #<issue number>(, Fixes #<issue_number>, ...)` format, to close the issue(s) when PR gets merged.
  Use `Partially addresses #<issue number>` to link an issue without closing it when the PR merges.
  
- Fixes #
- Partially addresses #
-->
<!-- optional,
  VEP tracking issue if this PR is implementing one.
  For additional info about VEP tracking issue, see https://github.com/kubevirt/enhancements#process
  
- VEP tracking issue: https://github.com/kubevirt/enhancements/issues/<vep_tracking_issue_number>
-->

### Why we need it and why it was done in this way
The following tradeoffs were made:

The following alternatives were considered:

Links to places where the discussion took place: <!-- optional: slack, other GH issue, mailinglist, ... -->

### Special notes for your reviewer

<!-- optional -->

### Checklist

This checklist is not enforcing, but it's a reminder of items that could be relevant to every PR.
Approvers are expected to review this list.

- [ ] Design: A [design document](https://github.com/kubevirt/community/tree/main/design-proposals) was considered and is present (link) or not required
- [ ] PR: The PR description is expressive enough and will help future contributors
- [ ] Code: [Write code that humans can understand](https://en.wikiquote.org/wiki/Martin_Fowler#code-for-humans) and [Keep it simple](https://en.wikipedia.org/wiki/KISS_principle)
- [ ] Refactor: You have [left the code cleaner than you found it (Boy Scout Rule)](https://learning.oreilly.com/library/view/97-things-every/9780596809515/ch08.html)
- [ ] Upgrade: Impact of this change on upgrade flows was considered and addressed if required
- [ ] Testing: New code requires [new unit tests](https://github.com/kubevirt/kubevirt/blob/main/docs/reviewer-guide.md#when-is-a-pr-good-enough). New features and bug fixes require at least one e2e test
- [ ] Documentation: A [user-guide update](https://github.com/kubevirt/user-guide/) was considered and is present (link) or not required. You want a user-guide update if it's a user facing feature / API change.
- [ ] Community: Announcement to [kubevirt-dev](https://groups.google.com/g/kubevirt-dev/) was considered

### Release note
<!--  Write your release note:
1. Enter your extended release note in the below block. If the PR requires additional action from users switching to the new release, include the string "action required".
2. If no release note is required, just write "NONE".
-->
```release-note
NONE
```

nirdothan commented:
/cc @orelmisan @oshoval @EdDev

orelmisan commented on pkg/network/admitter/admit_suite_test.go:
```diff
@@ -43,7 +42,3 @@ func (s stubClusterConfigChecker) IsBridgeInterfaceOnPodNetworkEnabled() bool {
 	return s.macvtapFeatureGateEnabled
 }
-
-func (s stubClusterConfigChecker) PasstEnabled() bool {
```
There is also the production counterpart that could be removed:
https://github.com/kubevirt/kubevirt/blob/dc943111806d9a68f7b62a1e3979e6ff63b2d23f/pkg/virt-config/feature-gates.go#L120

nirdothan replied:
Leaving this to a follow-up PR, with possibly additional cleanups WRT Slirp.

nirdothan commented:
Added 2 commits blocking VMs from using DeprecatedPasst addressing @orelmisan's [comment](https://github.com/kubevirt/kubevirt/pull/16428#pullrequestreview-3617939193) and rebased.

orelmisan commented on pkg/network/admitter/discontinued.go:
```diff
@@ -37,6 +37,13 @@ func validateBlockDiscontinuedBindings(field *k8sfield.Path, spec *v1.VirtualMac
 				Field:   field.Child("domain", "devices", "interfaces").Index(idx).Child("slirp").String(),
 			})
 		}
+		if ifaceSpec.InterfaceBindingMethod.DeprecatedPasst != nil {
```
Commit message: The old paast binding has been discontinued.

nirdothan replied:
Done.

nirdothan commented:
Latest push addresses https://github.com/kubevirt/kubevirt/pull/16428#issuecomment-3703431835 in 3rd commit message.
PR desc updated.

orelmisan commented on pkg/network/admitter/discontinued.go:
```diff
@@ -37,6 +37,13 @@ func validateDiscontinuedBindings(field *k8sfield.Path, spec *v1.VirtualMachineI
+				Message: "Passt network binding has been discontinued since v1.3. Use PasstBinding instead",
+				Field:   field.Child("domain", "devices", "interfaces").Index(idx).Child("passt").String(),
+			})
+		}
```
Commit message:
I recommend dropping:
> It is replaced by a new biding named passtBinding.

as this PR stands on its own, even without the new passt binding.

nirdothan replied:
Done.

orelmisan reviewed (commented):
Thank you for the changes.

orelmisan reviewed (commented):
Thank you @nirdothan!
/approve

nirdothan commented:
[Pushed](https://github.com/kubevirt/kubevirt/compare/7af59e78f0884b092e653baea9af6d279cf575d8..9b1ca999741b05e82df5e9c7917e6344965015db) commit that removes virt-handler code per https://github.com/kubevirt/kubevirt/pull/16428#issuecomment-3707855364.

nirdothan commented:
/hold 
@orelmisan please remove if you approve changes

orelmisan commented:
Thanks for the changes, @nirdothan. Could you please update the PR subject and description to reflect that the discontinued `passt` binding is entirely removed across the stack?

nirdothan commented:
> Thanks for the changes, @nirdothan. Could you please update the PR subject and description to reflect that the discontinued `passt` binding is entirely removed across the stack?

Done.

orelmisan commented:
/unhold

nirdothan commented:
/test pull-kubevirt-manifests
